tst/IceFunctionalityTest.cpp

```diff
@@ -184,26 +185,33 @@ TEST_F(IceFunctionalityTest, connectionListenerFunctionalityTest)
     THREAD_JOIN(routine1, NULL);
     THREAD_JOIN(routine2, NULL);
 
-    EXPECT_EQ(STATUS_SUCCESS, doubleListGetNodeCount(pConnectionListener->connectionList, &connectionCount));
+    connectionCount = pConnectionListener->socketCount;
     EXPECT_EQ(connectionCount, routine1CustomData.connectionToAdd + routine2CustomData.connectionToAdd);
 
     CHECK(STATUS_SUCCEEDED(
         createSocketConnection((KVS_IP_FAMILY_TYPE) localhost.family, KVS_SOCKET_PROTOCOL_UDP, &localhost, NULL, 0, NULL, 0, &pSocketConnection)));
     EXPECT_EQ(STATUS_SUCCESS, connectionListenerAddConnection(pConnectionListener, pSocketConnection));
 
-    EXPECT_EQ(STATUS_SUCCESS, doubleListGetNodeCount(pConnectionListener->connectionList, &newConnectionCount));
+    newConnectionCount = pConnectionListener->socketCount;
     EXPECT_EQ(connectionCount + 1, newConnectionCount);
 
     EXPECT_EQ(STATUS_SUCCESS, connectionListenerRemoveConnection(pConnectionListener, pSocketConnection));
-    EXPECT_EQ(STATUS_SUCCESS, doubleListGetNodeCount(pConnectionListener->connectionList, &newConnectionCount));
+    newConnectionCount = pConnectionListener->socketCount;
     EXPECT_EQ(connectionCount, newConnectionCount);
 
-    EXPECT_EQ(TRUE, IS_VALID_TID_VALUE(pConnectionListener->receiveDataRoutine));
+    // Keeping TSAN happy need to lock/unlock when retrieving the value of TID
+    MUTEX_LOCK(pConnectionListener->lock);
+    threadId = pConnectionListener->receiveDataRoutine;
+    MUTEX_UNLOCK(pConnectionListener->lock);
+    EXPECT_TRUE( IS_VALID_TID_VALUE(threadId));
     ATOMIC_STORE_BOOL(&pConnectionListener->terminate, TRUE);
 
-    THREAD_SLEEP((SOCKET_WAIT_FOR_DATA_TIMEOUT_SECONDS + 1) * HUNDREDS_OF_NANOS_IN_A_SECOND);
+    THREAD_SLEEP(CONNECTION_LISTENER_SOCKET_WAIT_FOR_DATA_TIMEOUT + 1 * HUNDREDS_OF_NANOS_IN_A_SECOND);
```

Comment:
Previously sleep time was 2s. Now it is 1.2s.  Is this intended?

Reply:
Yeah, my mistake. Fixed